File: src/main/java/com/saybetter/domain/member/domain/Connect.java
package com.saybetter.domain.member.domain;

import com.saybetter.global.common.entity.BaseTimeEntity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Getter
@Entity(name = "CONNECT")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Connect extends BaseTimeEntity {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "connect_id")
	private Long connectId;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "educator_id")
	private Member educator;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "learner_id")
	private Member learner;
	
}

File: storage/storage-mysql-kt/src/main/kotlin/io/say/better/storage/mysql/domains/account/entity/Connect.kt
package io.say.better.storage.mysql.domains.account.entity

import io.say.better.storage.mysql.common.model.BaseTimeEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne

@Entity(name = "CONNECT")
class Connect(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "connect_id")
    private val connectId: Long = 0,
    @JoinColumn(name = "educator_id")
    @ManyToOne(fetch = FetchType.LAZY)
    private val educator: Educator,
    @JoinColumn(name = "learner_id")
    @ManyToOne(fetch = FetchType.LAZY)
    private val learner: Learner,
) : BaseTimeEntity()
